src/meta/src/manager/event_log.rs

```diff
@@ -0,0 +1,295 @@
+        enabled: bool,
+    ) -> Self {
+        if !enabled {
+            tracing::info!("event log is disabled");
+        }
+        Self {
+            meta_store,
+            event_tx,
+            worker_join_handle: RwLock::new(Some(worker_join_handle)),
+            enabled,
+        }
+    }
+
+    #[cfg(any(test, feature = "test"))]
+    pub fn for_test(meta_store: MetaStoreRef) -> Self {
+        let (event_tx, _event_rx) = tokio::sync::mpsc::channel(BUFFER_SIZE);
+        Self {
+            meta_store,
+            event_tx,
+            worker_join_handle: Default::default(),
+            enabled: false,
+        }
+    }
+
+    pub fn take_join_handle(&self) -> Option<(JoinHandle<()>, ShutdownSender)> {
+        self.worker_join_handle.write().take()
+    }
+
+    pub fn add_event_logs(&self, event_logs: Vec<EventLog>) -> MetaResult<()> {
+        if !self.enabled {
+            for event_log in event_logs {
+                tracing::debug!("{}", display_event_log(&event_log));
+            }
+            return Ok(());
+        }
+        let mut all_succ = true;
+        let processing_ts = SystemTime::now()
+            .duration_since(SystemTime::UNIX_EPOCH)
+            .unwrap()
+            .as_millis() as u64;
+        for mut event_log in event_logs {
+            event_log.unique_id = Some(uuid::Uuid::new_v4().to_string());
+            event_log.timestamp = Some(processing_ts);
+            // Intentionally drop event if any error of buffer is full.
+            if let Err(e) = self.event_tx.try_send(event_log.clone()) {
+                tracing::warn!(
+                    "event log is not persisted: {}, error {}",
+                    display_event_log(&event_log),
+                    e
+                );
+                all_succ = false;
+            }
+        }
+        if all_succ {
+            Ok(())
+        } else {
+            Err(anyhow::anyhow!("some event logs failed to persist").into())
+        }
+    }
+
+    pub async fn list_event_logs(&self) -> MetaResult<Vec<EventLog>> {
+        use prost::Message;
+        let event_logs = match as_etcd_client(&self.meta_store) {
+            None => vec![],
+            Some(c) => list(c)
+                .await
+                .map(|v| {
+                    v.into_iter()
+                        .map(|e| EventLog::decode(e.as_slice()).unwrap())
+                        .collect()
+                })
+                .map_err(|e| anyhow::anyhow!("failed to list event logs: {}", e))?,
+        };
+        Ok(event_logs)
+    }
+}
+
+fn display_event_log(event_log: &EventLog) -> String {
+    format!(
+        "id: {}, timestamp: {}, type: {}, info: {}",
+        event_log.unique_id.to_owned().unwrap_or_default(),
+        event_log.timestamp.to_owned().unwrap_or_default(),
+        event_log.event_type().as_str_name(),
+        event_log.info
+    )
+}
+
+pub fn new_event_log(event_type: EventType, info: String) -> EventLog {
```

Comment:
Consider making the log event strong-typed instead of JSON. For example


```suggestion
pub fn new_event_log(event: &EventLog) -> EventLog {
```

@yezizp2012 is now working on migrating from etcd to relational DB, so there should not be any JSON during the whole procedure.

Reply:
We can make `info` a structured proto message as follows:
```proto
  message EventInfo {
    uint32 job_id = 1;
    string name = 2;
    string definition = 3;
    string error = 4;
  }
```
And impl Display for it using macro `json!`.

Reply:
> We can make info a structured proto message as follows:

This message is specific to event log with type CREATE_STREAM_JOB_FAIL. Do we define a message for each type?

Reply:
If the event type is finite and added on a case-by-case basis, I think so.
But making `info` as `string` is also acceptable to me. We still can transform the text into json type for further analyze.